src/app/models/query.spec.ts

```diff
@@ -128,8 +138,285 @@ describe('function: parseQueryToString', () => {
 
 			const expectedResult = 'foo near:bar since:2017-06-22_04:25 until:2017-05-27_06:25';
 
-			const result: string = parseQueryToString(query);
+			const result: string = parseQueryToQueryString(query);
 
 			expect(result).toEqual(expectedResult);
 	});
 });
+
+describe('function: parseQueryToRouterString', () => {
+	it('should return string (foo) when no other attributes are set and displayString is foo', () => {
```

Comment:
could all these tests or chunks of these tests be done in one method with array of test cases? I am not sure. But could you check?